components/fl-operator/controllers/fledge_controller.go

```diff
@@ -47,16 +66,355 @@ type FlEdgeReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *FlEdgeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := logger.FromContext(ctx)
+	flEdge := &flv1alpha1.FlEdge{}
 
-	// TODO(user): your logic here
+	err := r.Get(ctx, req.NamespacedName, flEdge)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Error(err, "FlEdge resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
 
-	return ctrl.Result{}, nil
+		log.Error(err, "Could not get FlEdge")
+		return ctrl.Result{}, err
+	}
+
+	if flEdge.Spec.Auth.Spire != nil {
```

Comment:
A controller mutating its own spec as part of `Reconcile` is contrary to the purpose of a controller, which is ensuring that the state described in the spec matches the one of its managed resources.
This state could be handles as part of `FlEdge.Status` but I'm not sure if the spec would then have enough information to have a 1-to-1 mapping from spec to the manages resources.